kafka/modis_kernels_kf.py

```diff
@@ -229,21 +230,22 @@ def get_mcd43_prior(mcd43_fstring, band):
                 gdal.Open(os.path.join(the_dir, "brdf_2016_b06.vrt")),
                 gdal.Open(os.path.join(the_dir, "brdf_2016_b07.vrt")))
 
-    output_plethora = create_output_file(g, the_dir, "test")
+    output_plethora = create_output_file(g, the_dir, "band{}".format(band))
     
-    kf = MODISKernelLinearKalman(modis_obs, days, output_plethora, [] )
+    kf = MODISKernelLinearKalman(modis_obs, days, output_plethora, [], the_band=band )
```

Comment:
OK, so from L152, the `self.observations` object is
    
    doys qa vza sza saa vaa b01 b02 b03 b04 b05 b06 b07
     0   1   2   3   4   5   6   7
So **if** band is in MODIS order (1...7), then this is correct, **but** the band number must be decreased by one to get the relevant uncertainty in L132. I just can't quite remember what convention is used for `self.the_band`, but it appears to only be used in reading the data, so I don't think it matters greatly what convention is used, provided we stick to one.